torchqdynamics/ode/forward_solver.py

```diff
@@ -146,21 +148,18 @@ def _adaptive_odeint(self):
                 t, y, ft = t + dt, y_new, ft_new
 
                 # update the progress bar
-                pbar.update(dt.item())
-
-                # save results if flag is raised
-                if save_flag:
-                    self.save(y)
-
-            # return to the original dt, lower the flag and get next save time
-            if save_flag:
-                dt = dt_old
-                save_flag = False
-                next_tsave = self.next_tsave()
+                with warnings.catch_warnings():  # ignore tqdm precision overflow
+                    warnings.simplefilter('ignore', TqdmWarning)
+                    pbar.update(dt.item())
 
             # compute the next dt
+            if save_flag:  # return to the original dt
+                dt = dt_old
             dt = integrator.update_tstep(dt, error)
```

Comment:
So we might accept the `dt_old` step based on the `dt = self.next_tsave() - t` error.

Reply:
No I think it's fine. The bad thing to do would be to pass `dt = integrator.update_tstep(dt, error)` instead of `dt = integrator.update_tstep(dt_old, error)`, which might yield a really small dt and significantly slow down everything for no reason.

Here we're just passing a reduced `error` so dt might be a bit overestimated in the next step. But the update is capped by `max_factor` anyway. We could pass in a `error * dt_old / dt` instead of `error` if you'd like, that could be a bit better.

Reply:
But that might overshoot a lot the error in case dt_old / dt >> 1, so I'm not sure it's a good idea. Well worst case scenario the dt update is still capped.

Reply:
Ok ok I agree, it's fine like that, but maybe we could add a comment because it's a bit technical? Something along
```
# if `save_flag` is true the next time step is updated based on an underestimated error, so it might be overestimated, which is fine however as it is capped by `max_factor`
```